packages/components/segmented/src/segmented.vue

```diff
@@ -168,4 +181,24 @@ watch(
     flush: 'post',
   }
 )
+
+const updateDirection = () => {
+  if (props.direction === 'vertical' && groupRef.value) {
+    const groupElement = groupRef.value
+    groupElement.style.flexDirection = 'column'
+  } else if (groupRef.value) {
+    const groupElement = groupRef.value
+    groupElement.style.flexDirection = 'row'
+  }
+}
+onMounted(() => {
+  updateDirection()
+})
+
+watch(
+  () => props.direction,
+  () => {
+    updateDirection()
+  }
+)
```

Comment:
This part of the logic may be placed directly in the style, adding different classes according to different `direction` values, such as `el-segmented--vertical`/`el-segmented--horizontal`.